crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1500,6 +1499,14 @@ impl<'db> TypeInferenceBuilder<'db> {
         );
     }
 
+    pub(super) fn slice_step_size_zero_diagnostic(&mut self, node: AnyNodeRef) {
+        self.add_diagnostic(
+            node,
+            "slice-step-zero",
```

Comment:
Do we have a policy for choosing these names?

Reply:
Not yet. But it's probably going to follow Ruff's naming schema ([internal document](https://www.notion.so/astral-sh/Rule-Guidance-1324991853284188ae3bc15f08c07afa#482c66b35f6a4747a635d0066ff9dcad))